internal/native_image/common.bzl

```diff
@@ -0,0 +1,125 @@
+"Defines common properties shared by modern and legacy Native Image rules."
+
+load(
+    "//internal/native_image:builder.bzl",
+    _assemble_native_build_options = "assemble_native_build_options",
+)
+
+_RULES_REPO = "@rules_graalvm"
+_DEFAULT_GVM_REPO = "@graalvm"
+_GVM_TOOLCHAIN_TYPE = "%s//graalvm/toolchain" % _RULES_REPO
+_BAZEL_CPP_TOOLCHAIN_TYPE = "@bazel_tools//tools/cpp:toolchain_type"
+_BAZEL_CURRENT_CPP_TOOLCHAIN = "@bazel_tools//tools/cpp:current_cc_toolchain"
+_MACOS_CONSTRAINT = "@platforms//os:macos"
+_WINDOWS_CONSTRAINT = "@platforms//os:windows"
+
+_NATIVE_IMAGE_ATTRS = {
+    "deps": attr.label_list(
+        providers = [[JavaInfo]],
+        mandatory = True,
+    ),
+    "main_class": attr.string(
+        mandatory = False,
+    ),
+    "binary_type": attr.string(
+        mandatory = False,
+        default = "executable",
+    ),
```

Comment:
need to implement this or leave it out (it's intended to set up a shared library vs. executable build). downside of implementing now: it may need breaking changes to integrate with `java_binary` vs. `java_library`